pymatgen/entries/__init__.py

```diff
@@ -114,5 +142,25 @@ def as_dict(self) -> dict:
             "@module": self.__class__.__module__,
             "@class": self.__class__.__name__,
             "energy": self._energy,
-            "composition": self.composition.as_dict(),
+            "composition": self.composition.as_dict()
         }
+
+    def __eq__(self, other):
+        # NOTE Scaled duplicates i.e. physically equivalent materials
+        # are not equal unless normalized separately
+        if self is other:
+            return True
+
+        if isinstance(other, self.__class__):
+            # NOTE this is not performant and should be overwritten or
+            # preceeded if faster rigorous checks are available.
+            return self.as_dict() == other.as_dict()
```

Comment:
`as_dict()` also checks name and attribute for `PDEntry` which is why it is used here. In general I left that `as_dict()` was the best fail safe option as it will work for all `Entry` children. The only worry is that `as_dict()` of a `ComputedStructureEntry` is very slow hence the note about preceding with faster checks.

Reply:
Ok, i was worried that there might be a chance that two `as_dict`s might not be equal and that's very hard to troubleshoot. If you're confident that's not going to happen let's proceed with your version.